eng/scripts/TypeSpec-Requirement.ps1

```diff
@@ -98,27 +98,51 @@ else {
 
     try {
       $jsonContent = Get-Content $fullPath | ConvertFrom-Json -AsHashtable
+    }
+    catch {
+      LogWarning "  OpenAPI cannot be parsed as JSON, so assuming not generated from TypeSpec"
+      LogWarning "    $_"
+    }
 
+    if ($jsonContent) {
       if ($null -ne ${jsonContent}?["info"]?["x-typespec-generated"]) {
         LogInfo "  OpenAPI was generated from TypeSpec (contains '/info/x-typespec-generated')"
+
+        if ($file -match "specification/(?<specFamily>[^/]+)/") {
+          $specFamily = $Matches["specFamily"];
+          $tspConfigs = @(Get-ChildItem -Path (Join-Path $repoPath "specification" $specFamily) -Recurse -File
+            | Where-Object { $_.Name -eq "tspconfig.yaml" })
+
+          if ($tspConfigs) {
+            LogInfo "  Folder 'specification/$specFamily' contains $($tspConfigs.Count) file(s) named 'tspconfig.yaml'"
+          }
+          else {
+            LogError ("OpenAPI was generated from TypeSpec, but folder 'specification/$specFamily' contains no files name 'tspconfig.yaml'." `
+              + "  The TypeSpec used to generate OpenAPI must be added to this folder.")
```

Comment:
I understand you shorten `TypeSpec spec` and `OpenAPI spec` on purpose?

Like, full sentence would be:

> The TypeSpec spec used to generate OpenAPI spec must be added to this folder

Reply:
What error message would you use instead?  Consider that GitHub truncates after the words "used to generate" on my screen (see screenshot in PR description).

Reply:
I am just saying that I would replace `OpenAPI` with `OpenAPI spec` and similar with `TypeSpec`.